File: Dante1/Module1.02/Zadanie2.07/main.c

#include <stdio.h>

int main()
{
    double num1, num2, num3;

    printf("Enter number: ");

    int check = scanf("%lf", &num1);
    if (check != 1 || num1 < 0)
    {
        printf("Incorrect input");
        return 1;
    }

    if (num1 <= 50)
    {
        num2 = num1 * 0.50;
    }
    else if (num1 <= 150)
    {
        num2 = 50 * 0.50 + (num1 - 50) * 0.75;
    }
    else if (num1 <= 250)
    {
        num2 = 50 * 0.50 + 100 * 0.75 + (num1 - 150) * 1.20;
    }
    else
    {
        num2 = 50 * 0.50 + 100 * 0.75 + 100 * 1.20 + (num1 - 250) * 1.50;
    }

    num3 = num2 * (1 + 0.22);

    printf("Result is: %lf\n", num3);

    return 0;
}

File: Dante1/Module1.02/Zadanie2.07/main.rs
fn main()
{
    println!("Enter number: ");
    
    let mut number : String = String::new();
    std::io::stdin().read_line(&mut number).expect("Incorrect string input");
    
    let number: f32 = number.trim().parse().expect("Incorrect number parsing");
    
    let _number_2: f32;
    if number <= 50.0
    {
        _number_2 = number * 0.5;
    }
    else if number <= 150.0
    {
        _number_2 = 50.0 * 0.50 + (number - 50.0) * 0.75;
    }
    else if number <= 250.0
    {
        _number_2 = 50.0 * 0.50 + 100.0 * 0.75 + (number - 150.0) * 1.20;
    }
    else
    {
        _number_2 = 50.0 * 0.50 + 100.0 * 0.75 + 100.0 * 1.20 + (number - 250.0) * 1.50;
    }
    
    println!("Result is: {:.6}", _number_2 * (1.0 + 0.22));
    
    return;
}
